### PR TITLE
Linked list

### PR DESCRIPTION
**Pull request by lhchandler4**

### CODE REVIEW

**Comment by lhchandler4** [comment]
- [ X] Top-level README “Table of Contents” is updated
 - [ X] Feature tasks for this challenge are completed
 - [ X] Unit tests written and passing
     - [ X] “Happy Path” - Expected outcome
     - [ X] Expected failure
     - [ X] Edge Case (if applicable/obvious)
 - [ X] README for this challenge is complete
     - [ X] Summary, Description, Approach & Efficiency, Solution
     - [ X] Link to code
     - [ X] Picture of whiteboard

File: challenges/assets/README/LinkedList_README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 Linked Lists are a data structure that offer certain advantages over array lists
 
 ## Challenge
-The challenge was to create three methods. Insert, Includes, Print, Append, InsertBefore, InsertAfter
+The challenge was to create seven methods. Insert, Includes, Print, Append, InsertBefore, InsertAfter, InsertNFromTheEnd
 
 ## Approach & Efficiency
 Insert has a Big O(1). All the other methods have a Big O(n)
```
Comment (evanslaton) [inline challenges/assets/README/LinkedList_README.md:8]: I'm not seeing both the time and space efficiency!

File: challenges/src/main/java/Java/LinkedList/LinkedList.java [hunk 2]
```diff
@@ -96,4 +96,36 @@ public void insertAfter(int x, int data) {
             }
         }
     }
+
+    //
+    public boolean nothingIsThereBro(){
```
Comment (evanslaton) [inline challenges/src/main/java/Java/LinkedList/LinkedList.java:101]: #brogrammer

File: challenges/src/test/java/Java/LinkedList/LinkedListTest.java [hunk 3]
```diff
@@ -105,4 +105,42 @@ public void test_Insert_After() {
...
     }
 
+    @Test(expected = NullPointerException.class)
+    public void N_from_the_end_empty(){
+        LinkedList testing = new LinkedList();
+        testing.returnNfromTheEnd(0);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void test_N_from_end_k_is_negative(){
```
Comment (evanslaton) [inline challenges/src/test/java/Java/LinkedList/LinkedListTest.java:115]: `public void testKFromEnd_kIsNegative () {`

File: challenges/src/test/java/Java/LinkedList/LinkedListTest.java [hunk 4]
```diff
@@ -105,4 +105,42 @@ public void test_Insert_After() {
         assertEquals("1, 2, 4, 3, 6, 12, null", testing.prints());
     }
 
+    @Test(expected = NullPointerException.class)
+    public void N_from_the_end_empty(){
```
Comment (evanslaton) [inline challenges/src/test/java/Java/LinkedList/LinkedListTest.java:109]: `public void testKFromEnd_empty() {`
